lang/syn/src/parser/spl_interface.rs

```diff
@@ -0,0 +1,59 @@
+#[cfg(feature = "interface-instructions")]
+use syn::{Meta, NestedMeta, Path};
+
+#[cfg(not(feature = "interface-instructions"))]
+pub fn parse(_attrs: &[syn::Attribute]) -> Option<[u8; 8]> {
+    None
+}
+
+#[cfg(feature = "interface-instructions")]
+pub fn parse(attrs: &[syn::Attribute]) -> Option<[u8; 8]> {
+    let interfaces: Vec<[u8; 8]> = attrs
+        .iter()
+        .filter_map(|attr| {
+            if let Ok(Meta::List(meta_list)) = attr.parse_meta() {
+                if meta_list.path.is_ident("interface") {
+                    if let Some(NestedMeta::Meta(Meta::Path(path))) = meta_list.nested.first() {
+                        return Some(parse_interface_instruction(path));
+                    }
+                }
+            }
+            None
+        })
```

Comment:
I think this won't catch invalid inputs such as:

```rs
#[interface(spl_transfer_hook_interface:initialize_extra_account_meta_list)]
```

Reply:
Will investigate and add any necessary checks!

Reply:
Done!